pkg/oauth/oidc/interactive.go

```diff
@@ -45,8 +45,8 @@ func doOobFlow(cfg *oauth2.Config, stateToken string, opts []oauth2.AuthCodeOpti
 	return code
 }
 
```

Comment:
It's a valid case to have a caller set their own port, but this approach will break existing clients who are not providing a port. For them `oidcCallbackPort` will be `nil` which will cause the listen bind to fail.

I would suggest checking if a port value is passed in and if not, then use a random port. 

I expect would also need unit tests for this and the integration tests would need to be changed.

Reply:
i thought default value for int in go is 0. so without the flag, the behavior is same as before.

Reply:
I tested and it works like before.

Reply:
@lukehinds I agree we need to keep the default behaviour as it is. I mentioned here (https://github.com/sigstore/cosign/pull/1658#pullrequestreview-921547531) to add a flag with the redirect URI itself instead of just passing the port. There would be situations where localhost must not be ideal, so just exposing the port isn't ideal in my opinion.

Reply:
@hectorj2f agree, it makes sense we also allow a IP to be passed in. 

> do you mean this is a lib and function signature should not be changed ?

Yes, its used by cosign (iirc) and other libraries.